File: frontend/src/utils/severity.js
// 告警级别：统一为 critical / warning / info 三值，展示为 英文(中文)
export const SEVERITY_LABELS = {
  critical: 'Critical(严重)',
  warning: 'Warning(警告)',
  info: 'Info(提示)',
}

const SEVERITY_MAP = {
  critical: 'critical', fatal: 'critical', emergency: 'critical',
  severe: 'critical', error: 'critical', alert: 'critical', high: 'critical',
  warning: 'warning', warn: 'warning', medium: 'warning', moderate: 'warning',
  info: 'info', information: 'info', notice: 'info',
  low: 'info', ok: 'info', okay: 'info', normal: 'info', debug: 'info',
}

// 把任意来源的级别值归一为 critical/warning/info
export function normalizeSeverity(value) {
  if (!value) return 'warning'
  const key = String(value).trim().toLowerCase()
  if (SEVERITY_MAP[key]) return SEVERITY_MAP[key]
  if (/(critical|fatal|emergency|severe|error)/.test(key)) return 'critical'
  if (/(warn|medium|moderate)/.test(key)) return 'warning'
  if (/(info|notice|low|ok|normal|debug)/.test(key)) return 'info'
  return 'warning'
}

// 展示文本：Critical(严重) / Warning(警告) / Info(提示)
export function severityLabel(value) {
  return SEVERITY_LABELS[normalizeSeverity(value)]
}

File: app/models/severity.py
"""告警级别枚举与规范化

对外只暴露三种级别：critical / warning / info。
所有来源（外部告警、LLM、规则引擎）的级别值都统一归一到这里。
"""

SEVERITY_LEVELS = ("critical", "warning", "info")
SEVERITY_ZH = {"critical": "严重", "warning": "警告", "info": "提示"}

_SEVERITY_MAP = {
    "critical": "critical", "fatal": "critical", "emergency": "critical",
    "severe": "critical", "error": "critical", "alert": "critical", "high": "critical",
    "warning": "warning", "warn": "warning", "medium": "warning", "moderate": "warning",
    "info": "info", "information": "info", "notice": "info",
    "low": "info", "ok": "info", "okay": "info", "normal": "info", "debug": "info",
}

_WORDS_CRITICAL = ("critical", "fatal", "emergency", "severe", "error", "disaster")
_WORDS_WARNING = ("warn", "medium", "moderate", "attention")
_WORDS_INFO = ("info", "notice", "low", "ok", "normal", "debug", "recovered", "recovery")


def normalize_severity(value) -> str:
    """将任意来源的严重级别归一为 critical / warning / info 三值"""
    if not value:
        return "warning"
    key = str(value).strip().lower()
    if key in _SEVERITY_MAP:
        return _SEVERITY_MAP[key]
    if any(w in key for w in _WORDS_CRITICAL):
        return "critical"
    if any(w in key for w in _WORDS_WARNING):
        return "warning"
    if any(w in key for w in _WORDS_INFO):
        return "info"
    return "warning"
